File: ios/Classes/Data/MonitoringState.swift
//
//  Copyright (c) 2018 Loup Inc.
//  Licensed under Apache License v2.0
//

import Foundation
import CoreLocation

enum MonitoringState: String, Codable {
  case enterOrInside = "enterOrInside"
  case exitOrOutside = "exitOrOutside"
  case unknown = "unknown"
  
  init(from regionState: CLRegionState) {
    switch regionState {
    case .inside:
      self = MonitoringState.enterOrInside
    case .outside:
      self = MonitoringState.exitOrOutside
    case .unknown:
      self = MonitoringState.unknown
    }
  }
}

File: android/src/main/kotlin/io/intheloup/beacons/data/MonitoringState.kt
//  Copyright (c) 2018 Loup Inc.
//  Licensed under Apache License v2.0

package io.intheloup.beacons.data

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

enum class MonitoringState {
    EnterOrInside, ExitOrOutside, Unknown;

    class Adapter {
        @FromJson
        fun fromJson(json: String): MonitoringState =
                MonitoringState.valueOf(json.capitalize())

        @ToJson
        fun toJson(value: MonitoringState): String =
                value.toString().decapitalize()
    }
}